src/solver/simulation/solver.hxx

```diff
@@ -461,7 +469,7 @@ void ISimulation<Impl>::writeResults(bool synthesis, uint year, uint numSpace)
         }
 
         // Dumping
-        if (IO::Directory::Create(newPath))
+        if (IO::Directory::Create(newPath) && pResultWriter)
```

Comment:
As we know **pResultWriter** is not null (checked in **ISimulation<Impl>**'s constructor), do we need to test it here ? Shouldn't we only have : 
```C++
if (IO::Directory::Create(newPath))
{
    ...
}
```
?